Content.Server/Engineering/EntitySystems/DisassembleOnAltVerbSystem.cs

```diff
@@ -1,69 +1,24 @@
-using Content.Shared.DoAfter;
 using Content.Shared.Hands.EntitySystems;
-using Content.Shared.Verbs;
-using JetBrains.Annotations;
+using Content.Shared.Engineering.Components;
 
-namespace Content.Server.Engineering.EntitySystems
-{
-    [UsedImplicitly]
-    public sealed class DisassembleOnAltVerbSystem : EntitySystem
-    {
-        [Dependency] private readonly SharedHandsSystem _handsSystem = default!;
-
-        public override void Initialize()
-        {
-            base.Initialize();
-
-            SubscribeLocalEvent<DisassembleOnAltVerbComponent, GetVerbsEvent<AlternativeVerb>>(AddDisassembleVerb);
-        }
-        private void AddDisassembleVerb(EntityUid uid, DisassembleOnAltVerbComponent component, GetVerbsEvent<AlternativeVerb> args)
-        {
-            if (!args.CanInteract || !args.CanAccess || args.Hands == null)
-                return;
-
-            AlternativeVerb verb = new()
-            {
-                Act = () =>
-                {
-                    AttemptDisassemble(uid, args.User, args.Target, component);
-                },
-                Text = Loc.GetString("disassemble-system-verb-disassemble"),
-                Priority = 2
-            };
-            args.Verbs.Add(verb);
-        }
-
-        public async void AttemptDisassemble(EntityUid uid, EntityUid user, EntityUid target, DisassembleOnAltVerbComponent? component = null)
-        {
-            if (!Resolve(uid, ref component))
-                return;
-            if (string.IsNullOrEmpty(component.Prototype))
-                return;
-
-            if (component.DoAfterTime > 0 && TryGet<SharedDoAfterSystem>(out var doAfterSystem))
-            {
-                var doAfterArgs = new DoAfterArgs(EntityManager, user, component.DoAfterTime, new AwaitedDoAfterEvent(), null)
-                {
-                    BreakOnMove = true,
-                };
-                var result = await doAfterSystem.WaitDoAfter(doAfterArgs);
+namespace Content.Server.Engineering.EntitySystems;
 
-                if (result != DoAfterStatus.Finished)
-                    return;
-            }
-
-            if (component.Deleted || Deleted(uid))
-                return;
-
-            if (!TryComp(uid, out TransformComponent? transformComp))
-                return;
+public sealed class DisassembleOnAltVerbSystem : EntitySystem
+{
+    [Dependency] private readonly SharedHandsSystem _handsSystem = default!;
 
-            var entity = EntityManager.SpawnEntity(component.Prototype, transformComp.Coordinates);
+    public override void Initialize()
+    {
+        base.Initialize();
+        SubscribeLocalEvent<DisassembleOnAltVerbComponent, DissembleDoAfterEvent>(OnDissembleDoAfter);
+    }
 
-            _handsSystem.TryPickup(user, entity);
+    private void OnDissembleDoAfter(Entity<DisassembleOnAltVerbComponent> entity, ref DissembleDoAfterEvent args)
+    {
+        var spawnedEnt = SpawnNextToOrDrop(entity.Comp.PrototypeToSpawn, entity.Owner);
```

Comment:
Is this preferable over a simple `Spawn(...)`? Maybe if you want to disassemble something that is in the same container as you, but for structures this is irrelevant. Not sure about forks or so.